include/knowhere/index.h

```diff
@@ -117,90 +122,84 @@ class Index {
-        {
-            auto res = Config::FormatAndCheck(*cfg, json_);
-            LOG_KNOWHERE_DEBUG_ << "Add config dump: " << json_.dump();
-            if (res != Status::success) {
-                return res;
-            }
-        }
-        auto res = Config::Load(*cfg, json_, knowhere::TRAIN);
-        if (res != Status::success) {
-            return res;
-        }
-        return this->node->Add(dataset, *cfg);
+        KNOWHERE_CHECK_STATUS(Config::FormatAndCheck(*cfg, json_));
+        LOG_KNOWHERE_DEBUG_ << "Add config dump: " << json_.dump();
+        KNOWHERE_CHECK_STATUS(Config::Load(*cfg, json_, knowhere::TRAIN));
+
+        KNOWHERE_CHECK_STATUS(this->node->Add(dataset, *cfg));
+        return Status::success;
     }
 
     expected<DataSetPtr, Status>
     Search(const DataSet& dataset, const Json& json, const BitsetView& bitset) const {
         Json json_(json);
         auto cfg = this->node->CreateConfig();
-        {
-            auto res = Config::FormatAndCheck(*cfg, json_);
-            LOG_KNOWHERE_DEBUG_ << "Search config dump: " << json_.dump();
-            if (res != Status::success) {
-                return unexpected(res);
-            }
-        }
-        auto res = Config::Load(*cfg, json_, knowhere::SEARCH);
-        if (res != Status::success) {
-            return unexpected(res);
-        }
-        return this->node->Search(dataset, *cfg, bitset);
+        KNOWHERE_CHECK_EXPECTED(Config::FormatAndCheck(*cfg, json_));
+        LOG_KNOWHERE_DEBUG_ << "Search config dump: " << json_.dump();
+        KNOWHERE_CHECK_EXPECTED(Config::Load(*cfg, json_, knowhere::SEARCH));
+
+#ifdef NOT_COMPILE_FOR_SWIG
+        TimeRecorder rc("Search");
+        auto res = this->node->Search(dataset, *cfg, bitset);
+        auto span = rc.ElapseFromBegin("done");
+        span *= 0.001;  // convert to ms
+        kw_search_latency.Observe(span);
+#else
+        auto res = this->node->Search(dataset, *cfg, bitset);
+#endif
+        return res;
     }
 
     expected<DataSetPtr, Status>
     RangeSearch(const DataSet& dataset, const Json& json, const BitsetView& bitset) const {
         Json json_(json);
         auto cfg = this->node->CreateConfig();
-        {
-            auto res = Config::FormatAndCheck(*cfg, json_);
-            LOG_KNOWHERE_DEBUG_ << "RangeSearch config dump: " << json_.dump();
-            if (res != Status::success) {
-                return unexpected(res);
-            }
-        }
-        auto res = Config::Load(*cfg, json_, knowhere::RANGE_SEARCH);
-        if (res != Status::success) {
-            return unexpected(res);
-        }
-        return this->node->RangeSearch(dataset, *cfg, bitset);
+        KNOWHERE_CHECK_EXPECTED(Config::FormatAndCheck(*cfg, json_));
+        LOG_KNOWHERE_DEBUG_ << "RangeSearch config dump: " << json_.dump();
+        KNOWHERE_CHECK_EXPECTED(Config::Load(*cfg, json_, knowhere::RANGE_SEARCH));
+
+#ifdef NOT_COMPILE_FOR_SWIG
```

Comment:
Why need to have this

Reply:
To avoid compiling prometheus for python interface, which is now problematic as of I know